test/external/dataplane/connectivity.go

```diff
@@ -40,165 +40,179 @@ var (
 	simpleHTTPServerCommand = []string{"python", "-m", "SimpleHTTPServer", "80"}
 )
 
+type testParams struct {
+	Framework         *framework.Framework
+	ToEndpointType    tcp.EndpointType
+	Networking        framework.NetworkingType
+	Cluster           framework.ClusterIndex
+	ClusterScheduling framework.NetworkPodScheduling
+}
+
 var _ = Describe("[external-dataplane] Connectivity", func() {
 	f := framework.NewFramework("ext-dataplane")
 
-	It("should be able to connect from an external app to a pod in a cluster", func() {
-		if framework.TestContext.GlobalnetEnabled {
-			testGlobalNetExternalConnectivity(f)
-		} else {
-			testExternalConnectivity(f)
-		}
-	})
-})
+	var toEndpointType tcp.EndpointType
+	var networking framework.NetworkingType
+	var cluster framework.ClusterIndex
+
+	verifyInteraction := func(clusterScheduling framework.NetworkPodScheduling) {
+		It("should be able to connect from an external app to a pod in a cluster", func() {
+			if framework.TestContext.GlobalnetEnabled {
+				framework.Skipf("Globalnet is enabled, skipping the test...")
+				return
+			}
+
+			testExternalConnectivity(testParams{
+				Framework:         f,
+				ToEndpointType:    toEndpointType,
+				Networking:        networking,
+				Cluster:           cluster,
+				ClusterScheduling: clusterScheduling,
+			})
+		})
+	}
 
-func testExternalConnectivity(f *framework.Framework) {
-	externalClusterName := getExternalClusterName(framework.TestContext.ClusterIDs)
+	When("In an external-app-connected cluster", func() {
```

Comment:
When the test case is executed, AFAIU the sentence will not be correct as you are using When and starting the sentence with "In ...". 

@tpantelis can you please suggest?

Reply:
Changed to `When("connected from a gateway-cluster"`.  The word is the same to the ascii art. Any ideas to improve the wording are welcomed.